src/main.rs

```diff
@@ -63,8 +62,44 @@ impl IdentityMappedAddr {
     }
 }
 
+// Symbols defined in `linker.ld`
+extern {
+    static mmap_ent: usize;
+    static _memory_map: usize;
+    static _kib_kernel_size: usize;
+    static __page_table_start: usize;
+    static __page_table_end: usize;
+    static __bootloader_end: usize;
+    static __bootloader_start: usize;
+}
+
 #[no_mangle]
-pub extern "C" fn load_elf(
+pub unsafe extern "C" fn stage_4() -> ! {
+    asm!("mov bx, 0x0" :::: "intel");
+    asm!("mov ss, bx" :::: "intel"); // Set stack segment
```

Comment:
Could you merge these two into a single `asm!` statement? This ensures that they are executed right after each other. Also, we should add `bx` to the clobber list.